Flex editor updates

br648 opened the pull request:
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Editor updates covering:

1. Booking Rules
2. Location Groups
3. Locations
4. Location shapes

Commit:
refactor(Added explict GTFS-lib reference and editor controllers for booking rules and location grou

Commit:
refactor(Update to define if the loaded feed is flex): Update to set the flex value depending on the

Commit:
refactor(Added Bson ignore to flex param): Added Bson ignore to flex param so it is not picked up by

Commit:
refactor(Moved the @BsonIngore from param to setter): Updated FeedSource to move @BsonIgnore from th

Commit:
refactor(FeedSource): Added getter and setter methods for flex param

Commit:
refactor(FeedSource.java): Removed flex getter and setter

Removed flex getter and setter

Commit:
refactor(DataManager.java): Added editor controllers for location meta data and location shapes

Commit:
refactor(pom.xml): Bumped gtfs-lib version

Bumped gtfs-lib version

Commit:
Merge branch 'dev' into flex-editor-updates

Commit:
refactor(pom.xml): Bumped GTFS-lib version

Bumped gtfs-lib version which include pattern stop updates

Commit:
Merge branch 'dev' into flex-editor-updates

Commit:
Merge branch 'dev' into flex-editor-updates

miles-grant-ibigroup committed:
refactor(manager/DataManager): update tables to reflect gtfs-lib changes

Commit:
Merge branch 'flex-editor-updates' of https://github.com/ibi-group/datatools-server into flex-editor-updates

Commit:
Merge branch 'dev' into flex-editor-updates

miles-grant-ibigroup committed:
chore(deps): upgrade to latest gtfs-lib

Commit:
Merge branch 'flex-editor-updates' of https://github.com/ibi-group/datatools-server into flex-editor-updates

Commit:
refactor(pom.xml): Bumped GTFS-lib version and updated snapshot matcher dependency source

miles-grant-ibigroup committed:
chore(deps): update gtfs-lib version

miles-grant-ibigroup committed:
chore(deps): update gtfs-lib

miles-grant-ibigroup committed:
chore(deps): update gtfs-lib

binh-dam-ibigroup reviewed (commented):
Could you advise on persisting the flex field after computing `presetFlex`, please?

binh-dam-ibigroup reviewed (changes requested):
See https://github.com/ibi-group/datatools-server/pull/439#pullrequestreview-866518230 and resolve failing tests, please.

Commit:
Merge branch 'dev' into flex-editor-updates

Commit:
refactor(DataManager.java): Re-ordered editor controllers to group flex updates

Commit:
refactor(Fixed merge conflicts): Fixed merge conflicts

Commit:
refactor(FeedSource.java): Added flex UI features param

br648 commented on src/main/java/com/conveyal/datatools/manager/models/FeedSource.java:
```diff
@@ -701,4 +712,11 @@ public FeedTransformRules getRulesForRetrievalMethod(FeedRetrievalMethod retriev
+     * Set the flex value according to the feed version loaded.
+     */
+    public void presetFlex(FeedVersion feedVersion) {
+        flex = feedVersion.feedLoadResult.isGTFSFlex();
```
Partially addressed with the inclusion of `flexUIFeaturesEnabled`.

br648 replied:
Addressed in 49db0e1689980059c9742bd25f17d641cb064099

Commit:
refactor(Included the latest commit of gtfs-lib and made a few changes in relation to this): Updated

miles-grant-ibigroup committed:
refactor(FeedVersion): Update feedSource flex setting on creation

br648 commented on src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java:
```diff
@@ -274,6 +274,12 @@ public void load(MonitorableJob.Status status, boolean isNewVersion) {
+
+
+            FeedSource flexSetFeedSource = this.parentFeedSource();
+            flexSetFeedSource.flex = this.feedLoadResult.isGTFSFlex();
```
@miles-grant-ibigroup is feedLoadResult guaranteed not to be null at this point? I had to add a check for this in `FeedSource#presetFlex`.

miles-grant-ibigroup replied:
I've made a corresponding gtfs-lib commit that always returns true or false https://github.com/conveyal/gtfs-lib/pull/335/commits/e4d5e611f14aceefc06676756c8f70e4c2180473

br648 replied:
@miles-grant-ibigroup It is still possible for `feedLoadResult` to be null and therefore fail when a call is made to `isGTFSFlex()` method.

miles-grant-ibigroup replied:
Thanks for looking into and fixing that!

Commit:
refactor(Updated snapshots to include null references): Update to work with the latest version of sn

Commit:
refactor(Removed snapshots that should not have been commited): Removed redundant snapshots

Commit:
Merge branch 'flex-editor-updates' of https://github.com/ibi-group/datatools-server into flex-editor-updates

Commit:
refactor(Update to include correct snapshot matcher and update to failing unit tests): Correct issue

Commit:
refactor(Bumped GTFS-lib version and address table ref name change): Updated tableRerence to tableRe

Commit:
refactor(Updated lib dependencies): Update snapshot and gtfs-lib versions and removed redundant null

Commit:
refactor(FeedSource.java): Reinstated null check on feedLoadResult

Commit:
refactor(FeedVersion.java): Added null check

Added null check on feed load result

miles-grant-ibigroup committed:
chore(deps): upgrade gtfs-lib

philip-cline committed:
chore(deps): upgrade gtfs-lib

philip-cline committed:
chore(repos): add snapshot repository for opengeo

miles-grant-ibigroup committed:
chore(deps): update opengeo repository

Commit:
refactor(pom.xml): Removed redundant repo and bumped GTFS-lib version

Commit:
refactor(Fixed merged conflict): Fixed merged conflict with GTFS-lib version

miles-grant-ibigroup committed:
refactor(FeedSource): persist flex check results

Commit:
refactor(Fixed merge conflict): Fixed merge conflict

Commit:
refactor(pom.xml): Reverted gtfs-lib back to previously used version which has been overwritten by m

The latest commit with flex changes does not work, so using previous commit which does

philip-cline committed:
refactor(pom.xml): Bumped GTFS-lib version

Commit:
refactor(pom.xml): Bumped GTFS-Lib version

Commit:
refactor(Defining feed source flex value): If a feed version is flex set the feed source to always b

br648 commented:
@binh-dam-ibigroup reviews on this PR have stalled some what. I've now addressed this comment from you: https://github.com/ibi-group/datatools-server/pull/439#discussion_r794757400, so I think this is ready for another review.

binh-dam-ibigroup commented on src/main/java/com/conveyal/datatools/manager/models/FeedSource.java:
```diff
@@ -775,14 +774,18 @@ public <T extends FeedTransformation> List<T> getActiveTransformations(FeedVersi
+            Persistence.feedSources.updateField(this.id, "flex", true);
+        }
     }
+
```
Remove extra blank line.

Commit:
chore(pom.xml): Updated the gtfs-lib version

gtfs-lib version now references the commit at the point it was merged into dev-flex

Commit:
Merge branch 'dev' into flex-editor-updates

binh-dam-ibigroup reviewed (approved):
Things seem to work for the most part, great job!

Commit:
refactor(FeedSource.java): Removed removed extra blank line

miles-grant-ibigroup committed:
Merge branch 'dev' into flex-editor-updates

# Conflicts:
#	pom.xml

miles-grant-ibigroup committed:
Merge branch 'dev' into flex-editor-updates

# Conflicts:
#	pom.xml
